realtime/redisPublisher.js

```diff
@@ -105,76 +123,19 @@ function publishObject(inst, event, changedKeys, ignoreAttributes) {
  */
 function publishSample(sampleInst, subjectModel, event, aspectModel) {
   const eventType = event || getSampleEventType(sampleInst);
-
-  // will be over written when unwrapping json.stringified fields
-  const sample = sampleInst.get ? sampleInst.get() : sampleInst;
-  const nameParts = sample.name.split('|');
-  const subName = nameParts[0];
-  const aspName = nameParts[1];
-  let promisesArr = [];
-
-  // if redis cache is on, get subject and aspect from cache
-  if (featureToggles.isFeatureEnabled(constants.featureName)) {
-    const aspKey = redisStore.toKey('aspect', aspName);
-    const subKey = redisStore.toKey('subject', subName);
-    promisesArr = [
-      redisClient.hgetallAsync(aspKey),
-      redisClient.hgetallAsync(subKey),
-    ];
-  } else {
-    const subOpts = {
-      where: {
-        absolutePath: subName,
-      },
-    };
-    const aspOpts = {
-      where: {
-        name: aspName,
-      },
-    };
-
-    const getAspect = aspectModel ? aspectModel.findOne(aspOpts) :
-                              Promise.resolve(sample.aspect);
-    promisesArr = [getAspect, subjectModel.findOne(subOpts)];
-  }
-
-  return Promise.all(promisesArr)
-  .then((responses) => {
-    const asp = responses[0];
-    const sub = responses[1];
-
-    const aspect = asp.get ? asp.get() : asp;
-    delete aspect.writers;
-
-    // if field value is stringified, will parse the value.
-    sample.aspect = redisStore.arrayStringsToJson(aspect,
-      constants.fieldsToStringify.aspect);
-
-    if (sub) {
-
-      /*
-       *pass the sample instance to the publishObject function only if the
-       *aspect and subject are published
-       */
-      if (sample.aspect && sample.aspect.isPublished && sub.isPublished) {
-
-        // attach subject to the sample
-        const subject = sub.get ? sub.get() : sub;
-        sample.subject = redisStore.arrayStringsToJson(subject,
-          constants.fieldsToStringify.subject);
-
-        // attach absolutePath field to the sample
-        sample.absolutePath = subName;
-        publishObject(sample, eventType);
-      }
-    }
-
+  const useSampleStore =
+    featureToggles.isFeatureEnabled('enableRedisSampleStore');
+  return rtUtils.attachAspectSubject(sampleInst, useSampleStore, subjectModel,
+    aspectModel)
+  .then((sample) => {
+    publishObject(sample, eventType);
```

Comment:
so in publishSample we publish the entire sample, but elsewhere we only publish the partial sample, if the flag publish partial sample is true?